lib/puppet/pops/parser/lexer2.rb

```diff
@@ -213,7 +213,7 @@ def lex_unquoted_string(string, locator, escapes, interpolate)
     @scanner = StringScanner.new(string)
     @locator = locator || Puppet::Pops::Parser::Locator.locator(string, '')
     @lexing_context[:escapes] = escapes || UQ_ESCAPES
-    @lexing_context[:uq_slurp_pattern] = (interpolate || !escapes.empty?) ? SLURP_UQ_PATTERN : SLURP_ALL_PATTERN
+    @lexing_context[:uq_slurp_pattern] = interpolate ? (!escapes.include?('$') ? SLURP_UQNE_PATTERN : SLURP_UQ_PATTERN) : SLURP_ALL_PATTERN
```

Comment:
Inner condition can be simplified to `(escapes.include?('$') ? SLURP_UQ_PATTERN : SLURP_UQNE_PATTERN)`